### PR TITLE
Fix Istio CNI issues between helm and istioctl installs

### LINKED ISSUE #43632: Document how to use helm to install istio plus cni module
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
1. Add a per installation method guide to the CNI installation like it is eg. done at https://istio.io/latest/docs/setup/additional-setup/gateway/. This to allow a working setup of cni for helm users.
2. In addition it would be cool to add some generic hint on how to figure out the cni-bin dir. For people new to the material it's hard to figure out that this can be most likely determined on a node like this:
```
worker:/# ps -ef | grep cni
root      1234  4321  4 Feb17 ?        10:59:12 kubelet --cni-bin-dir=/var/vcap/jobs/kubelet/packages/cni/bin --container-runtime=docker --docker=unix:///var/vcap/sys/run/docker/docker.sock --docker-endpoint=unix:///var/vcap/sys/run/docker/docker.sock --event-qps=0 --kubeconfig=/var/vcap/jobs/kubelet/config/kubeconfig --network-plugin=cni --pod-infra-container-image=registry.tkg.vmware.run/pause:3.2 --root-dir=/var/vcap/data/kubelet --cloud-provider=vsphere --config=/var/vcap/jobs/kubelet/config/kubeletconfig.yml --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
root     11189 10309  0 15:26 pts/0    00:00:00 grep --color=auto cni
root     28818 28790  0 Feb21 ?        00:02:01 install-cni --log_output_level=default:info
```

**Describe alternatives you've considered**
Dropping the flux compliant deployment of istio via helm charts.

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**
Multiple issues point to https://istio.io/latest/docs/setup/additional-setup/cni/ and state that his is the complete setup guide on how to get CNI running. The problem is that there is only a guide on how to use istioctl to get it running. Usage of the helm chart that are now no longer marked as experimental in the docs is not covered. Some issues also point to the cni helm chart at https://artifacthub.io/packages/helm/istio-official/cni which neither explains how to get it running.
Amending a otherwise working istio helm installation with a ```helm install istio-cni istio/cni -n kube-system``` or a ```helm install istio-cni istio/cni -n istio-system```are not enough to get cni working.

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

We found a discrepancy between istioctl and helm installs.

Fixes https://github.com/istio/istio/issues/43632 and https://github.com/istio/istio/issues/45034

Superset of https://github.com/istio/istio/pull/44984
Releated to https://github.com/istio/istio/pull/45036

### CODE REVIEW

**Comment by jacob-delgado** [comment]
Also related to https://github.com/istio/istio.io/pull/13265 and https://github.com/istio/istio.io/pull/13217

File: manifests/profiles/openshift.yaml
Comment (jwendell) [inline manifests/profiles/openshift.yaml:22]: this requires changing the template to include `default/`

File: releasenotes/notes/fix-chained-cni-helm.yaml
Comment (howardjohn) [inline releasenotes/notes/fix-chained-cni-helm.yaml:9]: This is probably fine and overly pedantic, but nothing except documentation was _fixed_; this implies its a bug fix.

**Review by jwendell — CHANGES_REQUESTED** [review]
Have you tested this on openshift? it looks like it's broken.

Apparently it didn't apply the `k8s.v1.cni.cncf.io/networks` annotation to the app pod (tried with sleep sample)

File: manifests/charts/istio-control/istio-discovery/files/injection-template.yaml
Comment (rrobb-aspen) [inline manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:48]: You made this suggestion on my PR, do you want to include this here as well?
```suggestion
    {{- if not .Values.istio_cni.chained and eq .Values.cni.provider "multus" }}
```

File: manifests/charts/istiod-remote/files/injection-template.yaml
Comment (rrobb-aspen) [inline manifests/charts/istiod-remote/files/injection-template.yaml:48]: Same as istiod injection template?

**Review by brobb7 — COMMENTED** [review]
Pulling some suggested changes from PR https://github.com/istio/istio/pull/45036 before closing that PR

File: manifests/charts/istio-control/istio-discovery/files/injection-template.yaml
Comment (jacob-delgado) [reply manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:48]: No

**Comment by jacob-delgado** [comment]
> Have you tested this on openshift? it looks like it's broken.
> 
> Apparently it didn't apply the `k8s.v1.cni.cncf.io/networks` annotation to the app pod (tried with sleep sample)

How did you install? With helm or istioctl? With helm it's working fine. I'll try again with istioctl

See https://github.com/istio/istio.io/pull/13265

and notably my comment at https://github.com/istio/istio.io/pull/13265#issuecomment-1572516368

Helm and istioctl do not install the same and the documentation is poor in helping people use the CNI charts. For platforms like OCP it'll fail so users will need to diagnose and fix the issue if they are using Helm.

I'll get back to this PR once I'm able to spin up another OCP cluster. I've been doing the majority of my testing on kind with multus (see https://gist.github.com/jacob-delgado/bb3a08d21e13130e6b685df75ec54a10).

**Comment by jacob-delgado** [comment]
@jwendell I tested the changes withe @rrobb-aspen and we were able to get both Helm and istioctl working for OpenShift. Can we get another review?

File: manifests/profiles/openshift.yaml
Comment (jacob-delgado) [inline manifests/profiles/openshift.yaml:21]: We don't need `enabled`, but we needed `chained` due to how the operator code merges in differences between the CNI chart and the istio-discovery chart.

We can remove `enabled` if it would make it more clear.

File: manifests/charts/install-OpenShift.md
Comment (jwendell) [inline manifests/charts/install-OpenShift.md:28]: did you mean chained=false?

**Review by jwendell — COMMENTED** [review]
I deployed this on OpenShift and bookinfo worked. So, I have no objections in merging this.

My opinion is that having these setting duplicate is weird. For instance in the openshift profile we are setting cni=enabled in two places. I guess that fixing this implies a huge refactoring of the code?

File: manifests/charts/install-OpenShift.md
Comment (jacob-delgado) [inline manifests/charts/install-OpenShift.md:28]:
```suggestion
 helm install -n istio-system istio-17 manifests/charts/istio-control/istio-discovery --set istio_cni.enabled=true --set istio_cni.chained=false --set global.jwtPolicy=first-party-jwt 
```

Comment (jacob-delgado) [reply manifests/charts/install-OpenShift.md]: Yes to both. Merged.

**Comment by jacob-delgado** [comment]
> I deployed this on OpenShift and bookinfo worked. So, I have no objections in merging this.
> 
> My opinion is that having these setting duplicate is weird. For instance in the openshift profile we are setting cni=enabled in two places. I guess that fixing this implies a huge refactoring of the code?

I agree that it is confusing. There is code in istioctl to merge in these values. I figure I can fix that once I get this merged. That'll be a simpler fix more people can comment on.

**Review by jwendell — APPROVED** [review]

File: manifests/charts/install-OpenShift.md
Comment (howardjohn) [inline manifests/charts/install-OpenShift.md:28]: do we need `global.jwtPolicy=first-party-jwt`? hopefully not? Its not in OpenShift IOP profile

Comment (howardjohn) [reply manifests/charts/install-OpenShift.md:28]: Its also on a few other lines btw, if we do end up dropping it

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/install-OpenShift.md
Comment (jacob-delgado) [reply manifests/charts/install-OpenShift.md:28]: I don't think we do. I don't have my openshift cluster up and running anymore. I'll try it the next time I do.

**Review by hanxiaop — APPROVED** [review]

File: releasenotes/notes/fix-chained-cni-helm.yaml
Comment (hanxiaop) [inline releasenotes/notes/fix-chained-cni-helm.yaml:2]: Should this be `bug-fix` instead of `feature`?

**Pull request merged**
